### DISCUSSION TITLE
Data sources for electricity price data

### DISCUSSION THREAD

**Discussion by pierresegonne** [electricitymaps/electricitymaps-contrib#4487] [Q&A]
# Data sources for electricity price data

The inclusion of electricity price data for end consumers would be a large milestone for expanding Electricity Maps' data coverage. We need your help to identify potential open and free data sources that could enable that data to appear on Electricity Maps. 🙏

### But doesn't Electricity Maps already show electricity prices?

Yes that's true! As you can see below on Fig. 1, Electricity Maps already displays electricity prices for some zones. 

<p align="center">
    <img width="433" alt="Screenshot 2022-08-24 at 09 51 32" src="https://user-images.githubusercontent.com/32778266/186362144-d5aea11a-3a02-4c8f-b401-c74ad02499c6.png">
</p>
<p align="center">
    <b>Figure 1</b>: Example of Electricity Maps' price data for 24h
</p>

The price data shown is nevertheless not representative of what end consumers would actually pay for their consumption of electricity. That price corresponds to the day-ahead price for electricity. That is the price that is decided a day ahead in advance for a MWh of electricity. That price is decided on a market and corresponds to what producers would receive in exchange for injecting that amount of electricity on the grid.
For that electricity to reach end consumers, other parties and processes must be accounted for. These include for example transmission systems and distribution systems.

### Decomposition of the price for end consumers

When actually accounting for everyhing that the end consumer must pay for to consume electricity we obtain something like what is displayed on Fig. 2.

<p align="center">
  <img width="600" alt="Screenshot 2022-08-24 at 09 52 02" src="https://user-images.githubusercontent.com/32778266/186362250-fd6b1a9e-4d21-4aac-bc5e-31d2837c06b6.png">
</p>
<p align="center">
    <b>Figure 2</b>: Typical decomposition of an electricity bill in DK.
</p>

This clearly shows that the price shown above, that corresponds to the electricity itself only represents around 40% (in DK) of the entire bill.
To be more exhaustive, we would therefore need more information, in Europe at least, regarding,

* Taxes
* Distribution costs (tarifs + subscriptions)
* Transmission costs
* Retailer subscription prices
* Public service obligations (= contribution to support renewable power installation)

Ideally, a data source would provide data about one or more of these points in a unified format for all EU countries.

**Comment by VIKTORVAV99** [comment]
Oh boy this will be a tricky one, at least for Sweden.

While the taxes and such are the same for the whole country the fees the utility takes varies a lot. This is further complicated by the fact you can and most do have to electricity utilities for each home. One is responsible for the last mile infrastructure and takes a fee for that, the other is the one you actually buy the electricity from.

Both of these take out a fee and for the utility that owns the infrastructure this usually varies per municipal or city at the very least.

The only way I see this working if the users can input some of the data themselves in the settings and we then calculate the price locally based on known factors like the day ahead price and taxes together with variable factors like utility fees.

**Comment by thomasgibon** [comment]
You can get annual statistics for each European country [here](https://ec.europa.eu/eurostat/databrowser/view/nrg_pc_204_c/default/table?lang=en).

I am assuming some taxes are per kWh (network costs, renewable premium...), and some are per € (VAT), so if you know the wholesale market price, you would be able to determine the rest dynamically?
![image](https://user-images.githubusercontent.com/23332955/186407645-619c1ef3-26b3-48e5-b850-1b8557a15685.png)

**Comment by baggiponte** [comment]
As far as Italy is concerned:

* [Terna](https://www.terna.it/en) is Italy's main electricity transmission system operator. They do publish some data [here](https://www.terna.it/en/electric-system/transparency-report), which I guess you already know.
  * They also maintain a [list of distribution companies](https://www.terna.it/en/electric-system/distribution-companies). The full list is [here](https://impresedistributrici.terna.it/DistributoriPublic.aspx?td=202cb962ac59075b964b07152d234b70) but is in Italian only.
* [ARERA](https://www.arera.it/it/index.htm#) is the Regulator for Electricity, Grids and the Environment (Autorità di Regolazione per Energia Reti e Ambiente) and they have [open data with prices](https://www.arera.it/it/prezzi.htm) at that link. I can dig deeper if needed!

**Comment by j0j0i996** [comment]
For Germany, [Zeit](https://www.zeit.de/wirtschaft/energiemonitor-deutschland-gaspreis-spritpreis-energieversorgung) has a great overview for the current price for new end customer contracts. They apparently use data from the German comparison portal [verivox](https://www.verivox.de/)

**Comment by mastrs** [comment]
I have source for Czechia as the network operator provides them, however, it's temporarily unusable as there's a not closer explained bug in the data provided. I'll add it if the CZ.pay parser will be approved when this is available again (raw electricity).
And for the other components of the total price it's little bit complicated as that's depending on several factors.

**Comment by pierresegonne** [comment]
Gathering here comments from Twitter / Linkedin:

Trying to aggregate by zone:


`AT`
<img width="602" alt="Screenshot 2022-08-25 at 10 53 51" src="https://user-images.githubusercontent.com/32778266/186620928-8ce2dd7e-0dab-46a8-af02-44ca49bc56ab.png">

`BE`
<img width="601" alt="Screenshot 2022-08-25 at 10 53 14" src="https://user-images.githubusercontent.com/32778266/186620796-33200e09-5dc7-473c-a130-7c976f89562d.png">

`DK`
* [Andel energi distribution tariffs](https://andelenergi.dk/sel/netpriser/)
* [Dansk Energi tool to find distributor](https://www.danskenergi.dk/vejledning/nettilslutning/find-netselskab)
* [Radius tariffs](https://radiuselnet.dk/elnetkunder/tariffer-og-netabonnement/?_gl=1*17y8pdz*_up*MQ..*_ga*MTk4MzA1OTMzNy4xNjYxMTc0Nzgz*_ga_TRQXB95J5R*MTY2MTE3NDc4Mi4xLjEuMTY2MTE3NDg1NC4wLjAuMA)

`ES`
<img width="413" alt="Screenshot 2022-08-25 at 10 57 16" src="https://user-images.githubusercontent.com/32778266/186621712-b0d900fa-cd0b-49c5-b9be-40761d95172c.png">


`FR`
<img width="605" alt="Screenshot 2022-08-25 at 10 55 15" src="https://user-images.githubusercontent.com/32778266/186621268-4df05781-3dfb-4737-8d58-dc7568682e43.png">
<img width="604" alt="Screenshot 2022-08-25 at 10 55 41" src="https://user-images.githubusercontent.com/32778266/186621383-7ea2980f-c567-4023
<img width="604" alt="Screenshot 2022-08-25 at 10 56 05" src="https://user-images.githubusercontent.com/32778266/186621448-1eb3545c-b44e-44d7-9c3f-4fe032b330e3.png">
-a439-710fb805a586.png">


`US`
<img width="420" alt="Screenshot 2022-08-25 at 10 56 28" src="https://user-images.githubusercontent.com/32778266/186621623-72477220-4fc7-4793-b6a6-0fbd97b9083b.png">


**General**
<img width="600" alt="Screenshot 2022-08-25 at 10 54 48" src="https://user-images.githubusercontent.com/32778266/186621173-0983e5ab-8c00-4d36-8be1-9b73318678f5.png">

**Comment by jarek** [comment]
In Ontario there are two different basic rateplans to choose from (time-of-use and flat-per-kWh): https://www.oeb.ca/consumer-information-and-protection/electricity-rates. Delivery fees vary by city or region (by operator of local distribution and billing), for example I pay (nominally) 40.70 CAD per 30 days, while my relatives pay 29.77 CAD per month. There are also various extra fees and credits. This is a recent bill for a 30 day period:

![image](https://user-images.githubusercontent.com/47415/189354451-b8fe0c9c-95d5-4281-96fe-7bd7f8359af2.png)

Not sure how you would process or display this...

**Reply by VIKTORVAV99** [reply]
Basically the same problem with Sweden, it would be easy to add the charges that are the same for everyone like taxes but the reginal fees would be pretty much impossible to display on the map without an extensive database and user input to select their specific region.

**Comment by alex-alontra** [comment]
In the case of Spain, which I'm familiar with, you can find the energy price with all charges included (transmission+distribution costs, ancillary services, utility margin, support to renewables, and the adjustment due to the current gas cap) at [https://www.esios.ree.es/en/pvpc](https://www.esios.ree.es/en/pvpc). However, these prices do not include taxes, which currently amount to 5.53% of the electricity price (27.19% before the measures taken by the government because of the war in Ukraine).

The final electricity bill also includes charges on the rated or maximum allowable power of the household but, since the escalation of prices, energy comprises the majority of the costs. You also need to know that only ~50% of the consumers in Spain pay the variable price or PVPC. The other 50% pay a fixed rate set by the utility company, though it is heavily based on the variable price.

**Comment by g1za** [comment]
For Italy:

Another useful source is Government GME website.
You can find a lot of data, day by day, (day-ahead starting from about 13:00 more or less), hour by hour (you can download compressed but also XML format) here:
[https://www.mercatoelettrico.org/It/download/DownloadDati.aspx?val=MGP_PrezziConvenzionali](url)

Also cumulative historical data here:
[https://www.mercatoelettrico.org/It/download/DatiStorici.aspx](url)

PUN, prices split by bidding areas, ... and even much more.
Keep in mind you are asked to accept the terms to access the data.

**Comment by corradio** [comment]
The following report gives a very good overview:
https://www.raponline.org/wp-content/uploads/2023/09/rap-jb-jh-smart-charging-europe-2022-april-26.pdf
